src/Components/Blazor/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -257,6 +255,8 @@
       <Output TaskParameter="Dependencies" ItemName="_BlazorResolvedRuntimeDependencies" />
     </ResolveBlazorRuntimeDependencies>
 
+    <WriteLinesToFile File="$(_BlazorApplicationAssembliesCacheFile)" Lines="@(_BlazorResolvedRuntimeDependencies)" Overwrite="true" />
```

Comment:
@pranavkm This is an entirely separate issue I noticed while looking at this. Previously, we never wrote the `unlinked.output` file to disk, which I guess is an unintentional omission. Adding this line means we do write it to disk.

However, it's unclear to me that this achieves anything. The "input" items for `_ResolveBlazorRuntimeDependencies` include `@(IntermediateAssembly)`, which is treated as changed on every build, even if you didn't modify your sources. As such, `_ResolveBlazorRuntimeDependencies` is never skipped, and hence we regenerate `unlinked.output` and `@(_BlazorResolvedRuntimeDependencies)` on every build.

So, what's the intention for `unlinked.output`? If we know the info will be recreated on every build, does that make it pointless to write it to disk?

Reply:
>  include @(IntermediateAssembly), which is treated as changed on every build, even if you didn't modify your sources.

That seems really suspect. I'll have a look at it, but the rest of the change here looks great.